akka-projection-alpakka-kafka/src/test/scala/akka/projection/scaladsl/kafka/ConsumerRecordAtMostOnceSpec.scala

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.scaladsl.kafka
+
+import akka.kafka.Subscriptions
+import akka.kafka.scaladsl.Consumer
+import akka.projection.testkit.{ TestEventHandler, TestInMemoryRepository }
+import akka.stream.testkit.scaladsl.StreamTestKit.assertAllStagesStopped
+import org.scalatest.concurrent.PatienceConfiguration.Timeout
+
+import scala.concurrent.duration._
+
+class ConsumerRecordAtMostOnceSpec extends TestcontainersKafkaSpec {
+
+  implicit val patience: PatienceConfig = PatienceConfig(30.seconds, 500.millis)
+
+  "Kafka Projection using at-most-once delivery" must {
+
+    "deliver events to ProjectionHandler" in assertAllStagesStopped {
+
+      val Messages = "abc" :: "def" :: "ghi" :: Nil
+      val topic1 = createTopic(1)
+      val group1 = createGroupId(1)
+
+      produceString(topic1, Messages).futureValue(Timeout(remainingOrDefault))
+
+      val repository = new TestInMemoryRepository[String]
+
+      val projection =
+        KafkaProjection(
+          Consumer.atMostOnceSource(consumerDefaults.withGroupId(group1), Subscriptions.topics(topic1)),
+          TestEventHandler(repository).asAsyncHandler)
```

Comment:
at-most-once example, but could be any Source returning ConsumerRecord. 

It's assumed that user will pass one that either has auto-commit or managed the commit in a at-most-once fashion. Failing to do so will produce a projection that start from scratch each time.